src/main/java/org/opensearch/sdk/rest/ExtensionDeprecationRestHandler.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.sdk.rest;
+
+import java.util.Objects;
+
+import org.opensearch.common.logging.DeprecationLogger;
+import org.opensearch.extensions.rest.ExtensionRestResponse;
+import org.opensearch.rest.DeprecationRestHandler;
+import org.opensearch.rest.RestRequest;
+
+/**
+ * {@code ExtensionDeprecationRestHandler} provides a proxy for any existing {@link Extension RestHandler} so that usage of the handler can be logged using the {@link DeprecationLogger}.
+ */
+public class ExtensionDeprecationRestHandler implements ExtensionRestHandler {
```

Comment:
We can have this class also as a nested class in `ExtensionRestHandler`

Reply:
Sure, this one makes sense.